scenarios/hci.yaml

```diff
@@ -0,0 +1,66 @@
+---
+undercloud:
+  config:
+    - section: DEFAULT
+      option: undercloud_hostname
+      value: undercloud.localdomain
+    - section: DEFAULT
+      option: undercloud_timezone
+      value: UTC
+    - section: DEFAULT
+      option: undercloud_debug
+      value: true
+    - section: DEFAULT
+      option: container_cli
+      value: podman
+    - section: DEFAULT
+      option: undercloud_enable_selinux
+      value: false
+    - section: DEFAULT
+      option: generate_service_certificate
+      value: false
+  undercloud_parameters_override: "hci/hieradata_overrides_undercloud.yaml"
+  undercloud_parameters_defaults: "hci/undercloud_parameter_defaults.yaml"
+  ctlplane_vip: 192.168.122.99
+pre_oc_run:
+  - name: Deploy Ceph
+    type: playbook
+    source: "adoption_deploy_ceph.yml"
+cloud_domain: "localdomain"
+hostname_groups_map:
+  # map ansible groups in the inventory to role hostname format for
+  # 17.1 deployment
+  osp-computes: "overcloud-computehci"
+  osp-controllers: "overcloud-controller"
+roles_groups_map:
+  # map ansible groups to tripleo Role names
+  osp-computes: "ComputeHCI"
+  osp-controllers: "Controller"
+overcloud:
+  stackname: "overcloud"
+  args:
+    - "--override-ansible-cfg /home/zuul/ansible_config.cfg"
+    - "--templates /usr/share/openstack-tripleo-heat-templates"
+    - "--libvirt-type qemu"
+    - "--timeout 90"
+    - "--overcloud-ssh-user zuul"
+    - "--deployed-server"
+    - "--validation-warnings-fatal"
+    - "--disable-validations"
+    - "--heat-type pod"
+    - "--disable-protected-resource-types"
+  vars:
+    - "/home/zuul/deployed_ceph.yaml"
+    - "/usr/share/openstack-tripleo-heat-templates/environments/docker-ha.yaml"
+    - "/usr/share/openstack-tripleo-heat-templates/environments/podman.yaml"
+    - "/usr/share/openstack-tripleo-heat-templates/environments/low-memory-usage.yaml"
+    - "/usr/share/openstack-tripleo-heat-templates/environments/debug.yaml"
+    - "/usr/share/openstack-tripleo-heat-templates/environments/manila-cephfsnative-config.yaml"
+    - "/usr/share/openstack-tripleo-heat-templates/environments/enable-legacy-telemetry.yaml"
+    - "/usr/share/openstack-tripleo-heat-templates/environments/cephadm/cephadm.yaml"
+    - "/usr/share/openstack-tripleo-heat-templates/environments/cephadm/ceph-mds.yaml"
```

Comment:
I wondered if we should get rid of `args` and `vars` and just put the whole deploy command as a variable to set. I mean, will the deployment command *always* be in the format that is defined in the other patch:

https://github.com/openstack-k8s-operators/ci-framework/pull/2297/files#diff-9e5f7c63638bbac9bfa4df077fe45e16b3acb56feaf89599aacce7a307624498R64

Same for the ceph deploy command, should we just export that out in full as a variable rather than have it in the other patch:

https://github.com/openstack-k8s-operators/ci-framework/pull/2297/files#diff-9a8face42b21d9562691b08533c92de7db4832e6314cb0b9c5479f42dd3d448cR131

In general i tend to be more in favor of exporting larger command chunks as variables perhaps due to being burned by Infrared in the past, where it tried to be very smart and compose the commands for us from various pieces of info, but we almost always ended up fighting it, trying to "trick it" to produce the commands we needed. So its "smartness" quite often stood in the way rather than helped.

I realize there is more complexity to this though, e.g. some of the overcloud deploy input files are generated inside the deployment role rather than provided from the scenario here, so what i'm suggesting above is perhaps easier said then done.

Maybe it would be best to stay the course and just have this as food for thought when designing/amending the `overcloud deploy`, `ceph deploy` and similar commands, that when we hardcode the structure for them, we should be hardcoding something quite un-opinionated.

As a last resort, we can introduce an override mechanism later in the form "If this variable is specified, use it as `overcloud deploy` command. If it is not specified, use the command hardcoded in the deployment role."

Reply:
I see what you mean, but as you mentioned dealing with the files generated dynamically will be tricky, I like your last suggestion though, to have an optional argument that'll allow the user to override the whole command, I'll add that and update both patches after testing